tools/ci_build/github/azure-pipelines/templates/py-linux.yml

```diff
@@ -106,3 +106,10 @@ jobs:
       inputs:
         artifactName: 'drop-linux-cpu-${{ parameters.arch }}'
         targetPath: '$(Build.BinariesDirectory)/${{ parameters.cmake_build_type }}'
+
+    - template: component-governance-component-detection-steps.yml
+      parameters :
+        condition : 'succeeded'
+    - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
```

Comment:
Please move the cleanup task as the first task.
Because the workspace is clean in the post task and post tasks are organized with stack.
That is, the last defined, the first executed